### PR TITLE
CLN: Use more memoryviews

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

File: pandas/_libs/reshape.pyx [hunk 1]
```diff
@@ -80,7 +80,7 @@ def unstack(numeric_object_t[:, :] values, const uint8_t[:] mask,
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
-def explode(ndarray[object] values):
```
Comment (jbrockmendel) [inline pandas/_libs/reshape.pyx:83]: can we get here with a readonly ndarray?

Comment (mroeschke) [reply pandas/_libs/reshape.pyx:83]: Looks like in our testing this array is always writable (from `mgr.internal_values()`), should this be readonly?

Comment (jbrockmendel) [reply pandas/_libs/reshape.pyx:83]: I think it can depend on what user passed to the constructor

Comment (mroeschke) [reply pandas/_libs/reshape.pyx:83]: Gotcha. Well specifically here, we can't use `const` as that's not supported with object memoryviews

**Review by WillAyd — APPROVED** [review]
lgtm

**Pull request merged**